src/Compilers/Core/Portable/Binding/UseSiteInfo.cs

```diff
@@ -526,7 +526,11 @@ public void InterlockedCompareExchange(TAssemblySymbol? primaryDependency, UseSi
             }
         }
 
-        public UseSiteInfo<TAssemblySymbol> InterlockedInitialize(TAssemblySymbol? primaryDependency, UseSiteInfo<TAssemblySymbol> value)
+        /// <summary>
+        /// Atomically initializes the cache with the given value if it is currently fully default.
+        /// This <i>will not</i> initialize <see cref="CachedUseSiteInfo{TAssemblySymbol}.Uninitialized"/>.
+        /// </summary>
+        public UseSiteInfo<TAssemblySymbol> InterlockedInitializeFromDefault(TAssemblySymbol? primaryDependency, UseSiteInfo<TAssemblySymbol> value)
```

Comment:
Note: the calling pattern in `PEMethodSymbol` is pretty complicated, and explicitly calls out not using the `Sentinel` for race safety. It seemed like it could be simplified to me, but I didn't want to touch it here, so I just renamed these methods to make it clear what they actually do: they both initialize a `CachedUseSiteInfo`, one assuming that `_info` is `null`, and one assuming that `info` is `Sentinel`.

Reply:
Thanks for the rename and this context. It is surprising indeed that we couldn't always use a sentinel for the uninitialized value.